homeassistant/components/bzutech/sensor.py

```diff
@@ -0,0 +1,373 @@
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="C10",
+        translation_key="pm10",
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="BAT",
+        translation_key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="DBM",
+        translation_key="dbm",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="MEM",
+        translation_key="memory",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="UPT",
+        translation_key="uptime",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+EndpointsSensors = {
+    "EP101": ["SHT20-TMP", "SHT20-HUM", "BH1750-LUM"],
+    "EP111": ["SHT20-TMP", "SHT20-HUM", "BH1750-LUM", "SHT30-TMP", "SHT30-HUM"],
+    "EP121": ["SHT20-TMP", "SHT20-HUM", "BH1750-LUM", "DOOR-DOR"],
+    "EP200": [
+        "SHT20-TMP",
+        "SHT20-HUM",
+        "SGP30-VOC",
+        "SGP30-CO2",
+        "SPS30-M01",
+        "SPS30-M25",
+        "SPS30-M40",
+        "SPS30-M10",
+    ],
+    "EP300": ["RTZSBZ-SND"],
+    "EP400": [
+        "ADS7878-VRA",
+        "ADS7878-VRB",
+        "ADS7878-VRC",
+        "ADS7878-CRA",
+        "ADS7878-CRB",
+        "ADS7878-CRC",
+        "ADS7878-CRN",
+        "ADS7878-APA",
+        "ADS7878-APB",
+        "ADS7878-APC",
+        "ADS7878-PPA",
+        "ADS7878-PPB",
+        "ADS7878-PPC",
+        "ADS7878-RPA",
+        "ADS7878-RPB",
+        "ADS7878-RPC",
+        "ADS7878-ACA",
+        "ADS7878-BCA",
+        "ADS7878-CCA",
+    ],
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Do entry Setup."""
+    bzu_api = hass.data[DOMAIN][entry.entry_id]
+    sensors = []
+
+    for sensor in EndpointsSensors[entry.data[CONF_ENDPOINT]]:
+        sense = sensor + "-" + entry.data[CONF_SENSORPORT]
+        for description in SENSOR_TYPE:
+            if description.key == sense.split("-")[1]:
+                sensors.append(
+                    BzuEntity(bzu_api, sense, entry, description=description)
+                )
+
+    async_add_entities(sensors, update_before_add=True)
+
+
+class BzuEntity(SensorEntity):
```

Comment:
Please review the [entity naming documentation](https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations) and set `has_entity_name` to True

Reply:
Did you get anywhere with this?